Adaptors/MongoDB/src/TaskTable.cs

```diff
@@ -526,6 +526,26 @@ await foreach (var taskId in taskCollection.AsQueryable(sessionHandle)
     }
   }
 
+  /// <inheritdoc />
+  public async Task RemoveRemainingDataDependenciesAsync(IEnumerable<(string taskId, IEnumerable<string> dependenciesToRemove)> dependencies,
```

Comment:
I am not sure we really need that much flexibility. From what I understand, we need only to remove the exact same results from all tasks.

```suggestion
  public async Task RemoveRemainingDataDependenciesAsync(ICollection<string> taskId, ICollection<string> dependenciesToRemove,
```

I think this would make the request lighter.

Reply:
As collections are not compatible with lists and arrays, its purpose is unclear to me

Reply:
My comment was not really about collection vs enumerable, but about the IEnumerable of IEnumerable. The interface you wrote is more flexible and enable to remove different sets of results depending on the task, but from what I understand, we always remove the same results from all the tasks. Thus, you can have 2 enumerables, one for results to remove, and the other for the tasks. The goal of this change is to make the db request smaller on the network, and less boilerplate to call this function.

Reply:
I disagree, it depends on the workflow of our users. Today we have mostly one consumer for our data. But it will change in the future, for instance in some algebra based applications. Thus, tasks may complete more than one result at a time and results can be used by more than one task. What you propose means that we will have to remove all the data dependencies from all the tasks even if we know that most of them are not in the list from which we will try to remove them.